BUILD.bazel

```diff
@@ -758,31 +758,56 @@ ray_cc_library(
 )
 
 ray_cc_library(
-    name = "stats_metric",
+    name = "stats_metric_core",
     srcs = [
         "src/ray/stats/metric.cc",
-        "src/ray/stats/metric_defs.cc",
-        "src/ray/stats/tag_defs.cc",
     ],
     hdrs = [
         "src/ray/stats/metric.h",
-        "src/ray/stats/metric_defs.h",
         "src/ray/stats/tag_defs.h",
     ],
     deps = [
         "//src/ray/util",
         "//src/ray/util:logging",
         "//src/ray/util:size_literals",
-        "@com_github_jupp0r_prometheus_cpp//pull",
         "@com_google_absl//absl/base:core_headers",
-        "@com_google_absl//absl/container:flat_hash_map",
-        "@com_google_absl//absl/memory",
-        "@com_google_absl//absl/strings",
-        "@com_google_googletest//:gtest",
-        "@io_opencensus_cpp//opencensus/exporters/stats/prometheus:prometheus_exporter",
-        "@io_opencensus_cpp//opencensus/exporters/stats/stdout:stdout_exporter",
+        "@io_opencensus_cpp//opencensus/stats:stats",
+    ],
+)
+
+ray_cc_library(
+    name = "stats_metric_defs",
+    srcs = [
+        "src/ray/stats/metric_defs.cc",
+        "src/ray/stats/tag_defs.cc",
+    ],
+    hdrs = [
+        "src/ray/stats/metric_defs.h",
+        "src/ray/stats/tag_defs.h",
+    ],
+    deps = [
+        ":stats_metric_core",
+    ],
```

Comment:
Should declare size_literals as dependency also: https://github.com/ray-project/ray/blob/e18a59ab6513f8332ce2abf1ed765c48ff2dce8a/src/ray/stats/metric_defs.cc#L17

Reply:
nit: order of the dependencies